### ISSUE TITLE
Frequently experience dprint redownloading the plugins

### ISSUE THREAD

**Issue by vlovich** [dprint/dprint#618]
Version: dprint 0.34.1

When I run `dprint fmt`, frequently it'll try to redownload the plugins even though I haven't changed anything afaik. This is particularly annoying for example if I lose internet connectivity or have bad internet connectivity because there's no fallback to use previously downloaded stuff.

Not sure if there's a log file I can attach to help debug or something.

**Comment by vlovich** [comment]
Running `touch dprint.json` can trigger this although there may be other things.

**Comment by dsherret** [comment]
That's weird. What operating system are you using? `dprint fmt --verbose` will say what it's doing behind the scenes. There should be some lines like...

```
[VERBOSE] Reading file: V:\dprint\dprint.json
[VERBOSE] Reading file: C:\Users\david\AppData\Local\dprint\cache\plugins\dprint-plugin-typescript\0.79.0-x86_64.cached
[VERBOSE] Reading file: C:\Users\david\AppData\Local\dprint\cache\plugins\dprint-plugin-json\0.17.0-x86_64.cached
[VERBOSE] Reading file: C:\Users\david\AppData\Local\dprint\cache\plugins\dprint-plugin-markdown\0.15.1-x86_64.cached
[VERBOSE] Reading file: C:\Users\david\AppData\Local\dprint\cache\plugins\dprint-plugin-toml\0.5.4-x86_64.cached
[VERBOSE] Checking path exists: C:\Users\david\AppData\Local\dprint\cache\plugins\dprint-plugin-prettier\0.17.0\x86_64\dprint-plugin-prettier.exe
[VERBOSE] Checking path exists: C:\Users\david\AppData\Local\dprint\cache\plugins\dprint-plugin-exec\0.3.5\x86_64\dprint-plugin-exec.exe
```

...at the start. You're saying you're able to reliably reproduce this? Maybe something external is modifying your cache directory.

**Comment by dsherret** [comment]
Thinking about it more, perhaps maybe you have an older version of dprint running in an editor somewhere? Or are using multiple versions? Having multiple versions of dprint running might cause plugins to redownload. At the moment I haven't built better handling of that into the system because dprint is not 1.0 stable yet.

**Comment by vlovich** [comment]
Ah that must be it then. I must have accidentally run my global dprint (0.19.1) and my local dprint (0.34.1). Will remedy that on my end and look forward to a day when it's solved. Thanks for the quick response. Please feel free to close as resolved or keep open to track the issue for 1.0. Amazing utility by the way.
